### ISSUE TITLE
itvhub download stopped working

### ISSUE THREAD

**Issue by brianmassen** [ytdl-org/youtube-dl#31363]
When I try to download from itvhub now I get this response
```
$ yt-dlp https://www.itv.com/watch/the-amazing-spider-man-2/2a3878
[generic] 2a3878: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 2a3878: Extracting information
ERROR: Unsupported URL: https://www.itv.com/watch/the-amazing-spider-man-2/2a3878
```
everything was fine until today.

Im running MX-Linux and yt-dlp version
```
$ yt-dlp -U
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
```
TIA

Brian

**Comment by MysticMerlin** [comment]
I had this problem late last night (Thursday 17/11/2022) & I was about to report this problem when I saw that you had reported it already Brian. The ITV Hub has now changed to ITVX so I'm assuming that the changes have caused this problem. As I tried downloading the News at 10 & couldn't I have only until midnight tonight to download it otherwise I can't. So, the urgency to get a fix is important to me for that reason. I'm also using the yt-dlp version which is up to date.

**Comment by MysticMerlin** [comment]
I forgot to say that the command line syntax I normally use & worked on Wednesday night is:-
yt-dlp https://www.itv.com/hub/itv-news-at-ten/2a4409a1812

The link has now changed to the following but as said it fails:-
yt-dlp https://www.itv.com/watch/itv-news-at-ten/2a4409/2a4409a1812

As you can see that it has watch & not hub in the address as well as extra set of numbers.
I haven't done a Verbose so I'll go & do that now.

**Comment by MysticMerlin** [comment]
![Screenshot (3554)](https://user-images.githubusercontent.com/25408759/202739306-c0a56068-8ea8-4629-b24e-0eef332e7418.png)
This is as you can see a screenshot of the verbose.

**Comment by Jonnyjon869** [comment]
@MysticMerlin some ITV programs (including ITN news) are available at https://www.stv.tv/ (the scottish version of ITV), and youtube-dl currently still supports this, so you can use that site to download some of the programmes you want to watch until a fix is found for ITV hub.

**Comment by october262** [comment]
> When I try to download from itvhub now I get this response
> 
> $ yt-dlp https://www.itv.com/watch/the-amazing-spider-man-2/2a3878 [generic] 2a3878: Downloading webpage WARNING: [generic] Falling back on generic information extractor [generic] 2a3878: Extracting information ERROR: Unsupported URL: https://www.itv.com/watch/the-amazing-spider-man-2/2a3878
> 
> everything was fine until today.
> 
> Im running MX-Linux and yt-dlp version
> 
> $ yt-dlp -U Latest version: 2022.11.11, Current version: 2022.11.11 yt-dlp is up to date (2022.11.11)
> 
> TIA
> 
> Brian

try using the cookies.txt file when you sign in to your account

**Comment by brianmassen** [comment]
Thanks for the suggestion but I’ve never had to sign in to download from the itvhub and I’m afraid don’t know what to do with  the cookies.txt file or where to get one from.

Brian

**Comment by october262** [comment]
> Thanks for the suggestion but I’ve never had to sign in to download from the itvhub and I’m afraid don’t know what to do with the cookies.txt file or where to get one from.
> 
> Brian

when i tried to watch that spiderman movie, it wanted me to setup an account
you can use either cookies from browser or the cookies.txt file for which you need to use a browser addon

**Comment by MysticMerlin** [comment]
@Jonnyjon869 Thanks for your suggestion Jonny & sorry I couldn't reply & try it earlier tonight. The English news programmes aren't available to download but other programmes like Loose Women & todays Loose Men as well as This Morning & Lorraine but not Good Morning Britain are available to download & do. So, your suggestion works & will work for other programmes that are available that I can continue to download until like you said until a fix has been done. Thanks again for your time in replying to me.

**Comment by dirkf** [comment]
All is explained in [the corresponding yt-dlp thread](https://github.com/yt-dlp/yt-dlp/issues/5570). The extractor code is basically the same.

tl;dr:
1. ITVX
2. don't hold your breath.

**Comment by MysticMerlin** [comment]
@dirkf Thanks for all your hard work & the others as well. When I did a search on here for the ITV problem, I can only recall that it was Brian's issue that I came across & commented etc. I've now looked at the links you've sent & I've got the gist of it all but obviously lost with the tech as I don't have the experience or knowledge to comment/help. So, I have to rely on you guys to come up with a fix. In the meantime, most of the programmes that I want I can get on the STV site with yt-dlp still working on there. Thanks Everyone.

**Comment by dirkf** [comment]
That's a useful work-around, as the STV pages have a completely different structure from ITV[Hub|-X].

**Comment by MysticMerlin** [comment]
@dirkf 
So, what does that mean then Dirk in the scheme of things especially youtube-dl?
Thanks.

**Comment by dirkf** [comment]
As you pointed out, what may not be extracted from ITV-X may be available from STV, and this applies equally for yt-dl. For instance, this evening's World Cup highlights from ITV (patched extractor that knows about the new URL format), using `--no-geo-bypass as I'm in the UK and setting a UA to avoid CloudFlare blocking:
```
$ python -m youtube_dl -F --no-geo-bypass --user-agent 'Mozilla/5.0' 'https://www.itv.com/watch/fifa-world-cup-2022-highlights/10a3625/10a3625a0005'
[ITV] 10a3625: Downloading webpage
[ITV] 10a3625: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
$
```
and from STV
```
$ youtube-dl -F --no-geo-bypass 'https://player.stv.tv/episode/4dgq/fifa-world-cup-highlights'
[stv:player] 4dgq: Downloading webpage
[brightcove:new] 6316229004112: Downloading JSON metadata
[brightcove:new] 6316229004112: Downloading m3u8 information
[brightcove:new] 6316229004112: Downloading m3u8 information
[brightcove:new] 6316229004112: Downloading m3u8 information
[brightcove:new] 6316229004112: Downloading m3u8 information
[brightcove:new] 6316229004112: Downloading MPD manifest
[brightcove:new] 6316229004112: Downloading MPD manifest
[brightcove:new] 6316229004112: Downloading MPD manifest
[brightcove:new] 6316229004112: Downloading MPD manifest
[info] Available formats for 6316229004112:
format code                                  extension  resolution note
hls-audio-0-en__Main_-0                      mp4        audio only [en] 
hls-audio-0-en__Main_-1                      mp4        audio only [en] 
hls-audio-1-en__Main_-0                      mp4        audio only [en] 
hls-audio-1-en__Main_-1                      mp4        audio only [en] 
hls-audio-2-en__Main_-0                      mp4        audio only [en] 
hls-audio-2-en__Main_-1                      mp4        audio only [en] 
dash-ec3d488d-dfa3-4d09-9054-4689bb76bcde-0  m4a        audio only [en] DASH audio   63k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-ec3d488d-dfa3-4d09-9054-4689bb76bcde-1  m4a        audio only [en] DASH audio   63k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-ec3d488d-dfa3-4d09-9054-4689bb76bcde-2  m4a        audio only [en] DASH audio   63k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-ec3d488d-dfa3-4d09-9054-4689bb76bcde-3  m4a        audio only [en] DASH audio   63k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-17bfad59-a0d6-4440-8991-eb3ae873b8bf-0  m4a        audio only [en] DASH audio   96k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-17bfad59-a0d6-4440-8991-eb3ae873b8bf-1  m4a        audio only [en] DASH audio   96k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-17bfad59-a0d6-4440-8991-eb3ae873b8bf-2  m4a        audio only [en] DASH audio   96k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-17bfad59-a0d6-4440-8991-eb3ae873b8bf-3  m4a        audio only [en] DASH audio   96k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-75346fe5-0408-40d3-a6d5-bcff69471592-0  m4a        audio only [en] DASH audio  127k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-75346fe5-0408-40d3-a6d5-bcff69471592-1  m4a        audio only [en] DASH audio  127k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-75346fe5-0408-40d3-a6d5-bcff69471592-2  m4a        audio only [en] DASH audio  127k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-75346fe5-0408-40d3-a6d5-bcff69471592-3  m4a        audio only [en] DASH audio  127k , m4a_dash container, mp4a.40.2 (44100Hz)
dash-8ea327e2-1d19-46e6-9ddb-59d9c9e1b6e1-0  mp4        320x180    DASH video  249k , mp4_dash container, avc1.42000c, video only
dash-8ea327e2-1d19-46e6-9ddb-59d9c9e1b6e1-1  mp4        320x180    DASH video  249k , mp4_dash container, avc1.42000c, video only
dash-8ea327e2-1d19-46e6-9ddb-59d9c9e1b6e1-2  mp4        320x180    DASH video  249k , mp4_dash container, avc1.42000c, video only
dash-8ea327e2-1d19-46e6-9ddb-59d9c9e1b6e1-3  mp4        320x180    DASH video  249k , mp4_dash container, avc1.42000c, video only
hls-343-0                                    mp4        320x180     343k , avc1.42000c, video only
hls-343-1                                    mp4        320x180     343k , avc1.42000c, video only
dash-03e0051f-eacc-468f-8468-d071754cc754-0  mp4        512x288    DASH video  474k , mp4_dash container, avc1.4d0015, video only
dash-03e0051f-eacc-468f-8468-d071754cc754-1  mp4        512x288    DASH video  474k , mp4_dash container, avc1.4d0015, video only
dash-03e0051f-eacc-468f-8468-d071754cc754-2  mp4        512x288    DASH video  474k , mp4_dash container, avc1.4d0015, video only
dash-03e0051f-eacc-468f-8468-d071754cc754-3  mp4        512x288    DASH video  474k , mp4_dash container, avc1.4d0015, video only
hls-627-0                                    mp4        512x288     627k , avc1.4d0015, video only
hls-627-1                                    mp4        512x288     627k , avc1.4d0015, video only
dash-70fd46e1-2bb1-49ed-a8bc-e20abf2e141e-0  mp4        768x432    DASH video  808k , mp4_dash container, avc1.4d001e, video only
dash-70fd46e1-2bb1-49ed-a8bc-e20abf2e141e-1  mp4        768x432    DASH video  808k , mp4_dash container, avc1.4d001e, video only
dash-70fd46e1-2bb1-49ed-a8bc-e20abf2e141e-2  mp4        768x432    DASH video  808k , mp4_dash container, avc1.4d001e, video only
dash-70fd46e1-2bb1-49ed-a8bc-e20abf2e141e-3  mp4        768x432    DASH video  808k , mp4_dash container, avc1.4d001e, video only
hls-1028-0                                   mp4        768x432    1028k , avc1.4d001e, video only
hls-1028-1                                   mp4        768x432    1028k , avc1.4d001e, video only
dash-845a8933-5195-44c7-ad5c-fc000c69cefa-0  mp4        1024x576   DASH video 1301k , mp4_dash container, avc1.4d001f, video only
dash-845a8933-5195-44c7-ad5c-fc000c69cefa-1  mp4        1024x576   DASH video 1301k , mp4_dash container, avc1.4d001f, video only
dash-845a8933-5195-44c7-ad5c-fc000c69cefa-2  mp4        1024x576   DASH video 1301k , mp4_dash container, avc1.4d001f, video only
dash-845a8933-5195-44c7-ad5c-fc000c69cefa-3  mp4        1024x576   DASH video 1301k , mp4_dash container, avc1.4d001f, video only
hls-1570-0                                   mp4        1024x576   1570k , avc1.4d001f, video only
hls-1570-1                                   mp4        1024x576   1570k , avc1.4d001f, video only
dash-bb1e7c0d-7e34-4551-85a4-6214b4518225-0  mp4        1280x720   DASH video 1953k , mp4_dash container, avc1.4d001f, video only
dash-bb1e7c0d-7e34-4551-85a4-6214b4518225-1  mp4        1280x720   DASH video 1953k , mp4_dash container, avc1.4d001f, video only
dash-bb1e7c0d-7e34-4551-85a4-6214b4518225-2  mp4        1280x720   DASH video 1953k , mp4_dash container, avc1.4d001f, video only
dash-bb1e7c0d-7e34-4551-85a4-6214b4518225-3  mp4        1280x720   DASH video 1953k , mp4_dash container, avc1.4d001f, video only
hls-2288-0                                   mp4        1280x720   2288k , avc1.4d001f, video only
hls-2288-1                                   mp4        1280x720   2288k , avc1.4d001f, video only
hls-343-2                                    mp4        320x180     343k , avc1.42000c, mp4a.40.2
hls-343-3                                    mp4        320x180     343k , avc1.42000c, mp4a.40.2
hls-627-2                                    mp4        512x288     627k , avc1.4d0015, mp4a.40.2
hls-627-3                                    mp4        512x288     627k , avc1.4d0015, mp4a.40.2
hls-1028-2                                   mp4        768x432    1028k , avc1.4d001e, mp4a.40.2
hls-1028-3                                   mp4        768x432    1028k , avc1.4d001e, mp4a.40.2
hls-1570-2                                   mp4        1024x576   1570k , avc1.4d001f, mp4a.40.2
hls-1570-3                                   mp4        1024x576   1570k , avc1.4d001f, mp4a.40.2
http-2123k-720p-0                            mp4        1280x720   2123k , MP4 container, H264, 739.89MiB
http-2123k-720p-1                            mp4        1280x720   2123k , MP4 container, H264, 739.89MiB
hls-2288-2                                   mp4        1280x720   2288k , avc1.4d001f, mp4a.40.2
hls-2288-3                                   mp4        1280x720   2288k , avc1.4d001f, mp4a.40.2 (best)
$
```
If the question is about a possible yt-dl ITV-X extractor, everything that is in the yt-dlp issue applies. If some magic can arrange for playable media to be found by yt-dlp, yt-dl will try to reproduce that.

**Comment by MysticMerlin** [comment]
@dirkf
Thank for that Dirk although I only grasp the basics above.
I think I've said previously & I think other people have as well that not all programmes are available from the STV site so a fix for the ITV Hub a.k.a. ITV-X is still required. I'm sorry if I'm quoting the obvious, but I'm only indicating that people like myself that can't code are relying a lot on those that can. So, thanks in advance for all the effort that those people do put in to youtube-dl etc.

**Comment by dirkf** [comment]
As noted [here](https://github.com/yt-dlp/yt-dlp/issues/5570#issuecomment-1342667369), now that ITVX has been launched officially (and without a `-`) the formats that could be found previously are still available, though probably not for Premium content (untested).

Here's an example from the yt-dlp thread accessed from the UK:
```
$ python -m youtube_dl -v -F 'https://www.itv.com/watch/family-guy/2a4259/2a4259a0399' --no-geo-bypass
[debug] System config: [u'--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'-F', u'https://www.itv.com/watch/family-guy/2a4259/2a4259a0399', u'--no-geo-bypass']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 774b346f9
[debug] Python version 2.7.18 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[ITV] 2a4259a0399: Downloading webpage
[ITV] 2a4259a0399: Downloading JSON metadata
[ITV] 2a4259a0399: Downloading m3u8 information
[info] Available formats for 2a4259a0399:
format code  extension  resolution note
hls-68       mp4        audio only   68k , mp4a.40.2@ 64k
hls-102      mp4        audio only  102k , mp4a.40.2@ 96k
hls-285      mp4        512x288     285k , avc1.4D401F@ 203k, 25.0fps, mp4a.40.2@ 64k
hls-636      mp4        512x288     636k , avc1.4D401F@ 503k, 25.0fps, mp4a.40.2@ 96k
hls-848      mp4        512x288     848k , avc1.4D401F@ 703k, 25.0fps, mp4a.40.2@ 96k
hls-1272     mp4        896x504    1272k , avc1.4D401F@1103k, 25.0fps, mp4a.40.2@ 96k
hls-1909     mp4        896x504    1909k , avc1.4D401F@1704k, 25.0fps, mp4a.40.2@ 96k (best)
$
```
The extractor will get updated in due course.

**Comment by Vangelis66** [comment]
> ` --no-geo-bypass` as I'm in the UK

Sadly, that flag (which provides an `X-Forwarded-For` request header on geo-sensitive APIs) **doesn't do anything these days** for "overseas" users, because both the playlist API (`magni.itv.com`) as well as ALL the stream CDNs: 

`itvpnpmobile.content.itv.com` (Akamai_1: 23.37.230.32, 23.37.230.42), 
`itvpnpmobile.blue.content.itv.com` (Akamai_2: 23.204.145.10, 23.204.145.19), 
`itvpnpmobile.cdn1.content.itv.com` (CloudFront: 13.35.125.20, 13.35.125.31, 13.35.125.54, 13.35.125.102)

**hard-geoblock** (i.e. can't be fooled by a spoofed `X-F-F` request header); thus, a **true whitelisted UK IP address** is required for geo-block circumvention; keeping the function inside `itv.py` does no harm, it simply doesn't work any more as designed (:disappointed:) ...

> and setting a UA **to avoid CloudFlare blocking**

I thought PR #30266 (specifically c6001d5) had already addressed that 😉 ; unless your test above did not use that PR's code (why not?) ...

Festive greetings 🎅 ...

**Comment by dirkf** [comment]
Good to know. We also need to set `_GEO_BYPASS = False` in the extractor to disable XFF and to detect the geo-block. What does the extractor report without VPN using `--no-geo-bypass`?

Although it would only affect the diagnostic, it would be interesting to know if a logged-in user can use the site from outside UK.

Maybe the answers to the above (`Outside the UK? ... you can only watch ITVX in the UK`, No) are [here](https://thevpn.guru/watch-itv-usa-unblock-uk-vpn-proxy).

I just patched the master code for the quoted test, but the changes will apply to the PR easily.

🎅 🎅

**Comment by MysticMerlin** [comment]
So, will we get a fix this year or next year? lol

**Comment by MysticMerlin** [comment]
I have only just seen that there has been an update to yt-dlp on the 06-01-2023 & I got my hopes up but sadly it hasn't solved downloading from the ITVX site. You can still download limited programmes from the STV site but like I said that you have limited programmes. For example one of my all time favourite programmes THE PRISONER is not available from the STV site but is from ITVX so I'm not happy obviously. Will we get a fix?

**Comment by dirkf** [comment]
Try the PR.

For those with an installation that exposes the source files, you can't just add the `extractor/itv.py` because it uses new routines from `utils.py`. Possibly updating both will do. If your source files are in a system directory (eg, `/usr/local/bin`), you have to run (eg) `youtube-dl --version` as admin after updating to update the compiled versions.

**Comment by MysticMerlin** [comment]
@dirkf
I'm sorry dirkf but I don't follow what you have said. I can tell you are very knowledgeable & experienced & even though I built my first PC in 1999 I haven't learnt everything & I've done very little coding. I learnt MSDOS to a degree before Windows so I like using what I call the DOS prompt.
So what do I have to do so that I can download from the ITVX site?
Thanks for your time & effort with the problems that arise. Like I've said that I'm not knowledgeable enough to contribute so I have to rely on people like you.

**Comment by dirkf** [comment]
You can wait for the PR to be merged and released, or ....
* install a Python version (if necessary) that supports yt-dl
* run `python -m pip "https://api.github.com/repos/ytdl-org/youtube-dl/zipball/pull/30266/head"`, where `python` is the name that invokes the Python executable (eg, maybe `python3` or `python3.10`)
* invoke the PR version of yt-dl with `youtube-dl ...` or possibly `python -m youtube_dl ...`

**Comment by MysticMerlin** [comment]
@dirkf
Thanks for your quick response dirkf.
I did view this quickly late last night but was too tired to take it in.
I've just checked that my version of Python on my laptop is 3.11.0.
I think I upgraded from Python 3.9.0 when the problem with the ITV arose.
The easy bit dirkf is to wait until the PR is merged and released but we don't know when that will be.
I don't know the name to run the Python executable. This is where I will sound thick now lol
I'm comfortable using the command line (I still call it the Dos Prompt lol)  but I'm a bit unsure of what to type including the correct syntax.
You said above to run this,
python -m pip "https://api.github.com/repos/ytdl-org/youtube-dl/zipball/pull/30266/head"
But I don't know what to put at the beginning instead of Python and do I leave out the quotation marks?
Also, the last bit about invoking the PR version of yt-dl with youtube-dl.

**Comment by dirkf** [comment]
`python --version` ?

**Comment by skovatch** [comment]
> @dirkf
> You said above to run this,
> python -m pip "https://api.github.com/repos/ytdl-org/youtube-dl/zipball/pull/30266/head"

First you need
`python -m pip install "https://api.github.com/repos/ytdl-org/youtube-dl/zipball/pull/30266/head"` 
and then 
`python -m youtube_dl --user-agent "Mozilla/5.0" -f mp4/best ....`
Note that's an '_', not '-'.

For me, running it this way causes the hlsnative extractor to delegate to ffmpeg, which is a bit annoying but still works.

**Comment by dirkf** [comment]
Now [use the git master](https://github.com/ytdl-org/youtube-dl/issues/31311#issuecomment-1301966817) instead.

The extractor suggests using the native HLS downloader but the M3U8 downloads from ITV require _ffmpeg_. yt-dl discovers this and complies.

**Comment by skovatch** [comment]
Yup, I didn't have pycryptodome installed... I thought I did but must have removed it at one from my python3 at one point.

**Comment by Vangelis66** [comment]
> For me, running it this way causes the **hlsnative** extractor to delegate to ffmpeg, which is a bit annoying

Hi Scott 😄 ; this has been detailed time and again 😉 ... 
The **mobile** `itvX` streams the itvIE "exploits" are of the **HLSe** type (Apple HLS **AES-128** encrypted); for [**hlsnative**] to decrypt them (without delegating dl+decryption to `ffmpeg`), you need one of `PyCrypto`, `Pycryptodome` **python modules** installed...

`PyCrypto` is old and unmaintained, however it can **still get the job done**... Main caveat being that **PyPi** doesn't host any wheels for it (v**2.6.1** stable and/or v**2.7a1**); the module has binary components (**C++** extensions, under **Windows OS** they have the `.pyd` file extension), so 

`python -m pip install -U pycrypto`

would require **compiling it from source**, which itself demands a working C++ compiler on your platform, compatible with the Python version used...

So your easiest avenue is to install the pycryptodome module: 

`python -m pip install -U pycryptodome`

**Edit**: As I was about to click the green `Comment` button, your latest post appeared 😜 ...

**Comment by MysticMerlin** [comment]
The obvious thing for me to do was wait for a new release or then do I try & follow the above instructions which I wasn't sure what I should do. So, I tried what skovatch said & I got the following you can see in the screenshots. B.T.W. Until I looed it up I didn't know what pip was except in the fruit & from the screenshot you can see that I've updated it.

![Screenshot (3814)](https://user-images.githubusercontent.com/25408759/218338332-3213a92e-aa30-4b47-b147-632b7501c4aa.png)

![Screenshot (3816)](https://user-images.githubusercontent.com/25408759/218338344-ed896dc5-220c-47c8-a344-987a90d6c209.png)

**Comment by Vangelis66** [comment]
@MysticMerlin 

The `itvX` "fix" has now (actually, on **Feb 3rd**) found its way onto the `master` branch: https://github.com/ytdl-org/youtube-dl/commit/384f632e8a9b61e864a26678d85b2b39933b9bae

Since it appears you're not "comfortable" running python scripts from source 😜 , please then grab and use the **Windows (32-bit) `yt-dl` executable** someone else was kind enough to compile for "Windows users": 

https://github.com/ytdl-patched/youtube-dl/releases/download/2023.02.12.43044/youtube-dl.exe

You **don't need to have Python installed** to run it, treat it like a "release" build you are used to...

**NB**: That build has no `Crypto` support, itself required by the `itvX` mobile, SD, streams; make sure you have a fairly recent `ffmpeg.exe` build adjacent to `youtube-dl.exe` (or in your `%PATH%`) ...

**Comment by MysticMerlin** [comment]
@Vangelis66 
Just seen your message at 13:50pm today. Thank you so much Vangelis for that. You've always been helpful in the past & now. I shall go & do what you say & hopefully get back to normal. And thanks to all that have contributed to the various software which like I've said I'm not experienced enough yet to contribute.

**Comment by Vangelis66** [comment]
The link to `youtube-dl.exe` in my [previous](https://github.com/ytdl-org/youtube-dl/issues/31363#issuecomment-1427164972) post has been **edited**, so that the most current build (at the time of downloading) is fetched... 

Also, and this is **important**, people here are **kindly asked not to abuse that "service"**, because it's basically a "personal" project that belongs to one of the `yt-dlp` contributors (many thanks, BTW 👍 ) ...

**Comment by MysticMerlin** [comment]
@Vangelis66
Hi again Van. Firstly, I hope it's okay with you by giving me pointers/help which hopefully can help other people like myself. The main reason why it's taken me longer than normal to try your suggestions etc is because I'm in the process of getting a better deal for car insurance as well as getting a better deal for a mobile phone contract etc which I'm trying to complete by the end of today. Before I get to the reason why I'm posting on here Van I don't think I've abused the service you mentioned & definitely not intentionally. Also I might sound thick when I say that I'm not sure how I can abuse the service.
In your first post I downloaded the youtube-dl file in question & I searched for the latest ffmpeg file. I downloaded (I think) the latest stable version as well as one that was posted/updated yesterday. Initially using that version I was getting errors downloading THE PRISIONER from the ITVX site which isn't available on the STV site & it didn't download. After being delayed trying to add the youtube-dl to the PATH & trying to get it to work so I just placed it in the same folder as ffmpeg etc, I changed the version of ffmpeg to version 4.0.2 & tried that. It works in a fashion as it does eventually download as you can see in the screenshot below.
So, is that how the downloads will be from now on with the timestamp errors or have I done something wrong?
Thanks in advance to you Vangelis66 & dirkf & all the others.
![Screenshot (3876)](https://user-images.githubusercontent.com/25408759/218724095-b2118b72-6d3b-4c99-a573-936b1810b9e0.png)

**Comment by MysticMerlin** [comment]
@Vangelis66
I'm sorry about this Van but here's something else.
Out of 14 episodes of CLASSIC EMMERDALE just 6 of them downloaded completely but still with the above timestamp error. The rest didn't download but had this error, ERROR: ffmpeg exited with code 1.
https://www.itv.com/watch/classic-emmerdale/2a8015/Ya0524a3402
https://www.itv.com/watch/classic-emmerdale/2a8015/Ya0524a3403
The first above downloaded & the second had the ffmpeg error.
See screenshot below, thanks.
![Screenshot (3882)](https://user-images.githubusercontent.com/25408759/218735551-3f46a666-09f7-4553-8867-4049cd41024d.png)

**Comment by dirkf** [comment]
Please open a new issue for https://www.itv.com/watch/classic-emmerdale/2a8015/Ya0524a3403 with a verbose log **posted as text**, if this persists.

I can't reproduce the 403, but maybe you displeased the ITV content serves somehow. `Invalid timestamp` errors are an artefact of _ffmpeg_ and shouldn't affect the payability of the merged result.

**Comment by dirkf** [comment]
See #32410.
